packages/client/src/store/selectors/attendance/attendanceVariance.ts

```diff
@@ -1,63 +1,88 @@
-import { SlotAttendnace, Customer } from "@eisbuk/shared";
+/* eslint-disable @typescript-eslint/ban-types */
+import {
+  type SlotAttendnace,
+  wrapIter,
+  keyMapper,
+  valueMapper,
+  mergeMapper,
+  ID,
+  _reduce,
+  CustomerAttendance,
+  SlotInterface,
+  SlotType,
+  flatMap,
+} from "@eisbuk/shared";
+import {
+  AthleteAttendanceMonth,
+  AttendanceByDate,
+  AttendanceBySlotType,
+  AttendanceDurations,
+  DateAttendancePair,
+} from "@eisbuk/ui";
 
 import { LocalStore } from "@/types/store";
+
 import { getMonthStr, calculateIntervalDuration } from "@/utils/helpers";
-import { type DayAttendanceHours } from "@eisbuk/ui";
-
-interface AttendanceRecord extends CustomerAttendanceRecord {
-  customerId: string;
-}
-
-interface CustomerAttendanceMap {
-  [customerId: string]: CustomerAttendanceRecord[];
-}
-
-interface CustomerAttendanceRecord {
-  date: string;
-  attendedInterval: number;
-  bookedInterval: number;
-}
-
-interface CustomerAttendanceRow {
-  athleteSurname: string;
-  athlete: string;
-  hours: DayAttendanceHours;
-}
-
-export const getMonthAttendanceVariance = (state: LocalStore) => {
-  const {
-    app: { calendarDay },
-    firestore: {
-      data: { attendance = [], customers = {} },
-    },
-  } = state;
 
-  const currentMonth = getMonthStr(calendarDay, 0);
+type CustomerNameTuple = [name: string, surname: string];
 
-  const customerAttendance = Object.values(attendance)
-    .filter(filterAttendanceByMonth(currentMonth))
-    .reduce(flattenAndConvertAttendanceIntervals, [])
-    .reduce(collectAttendanceByCustomer, {});
+type AttendanceDurationsWithType = AttendanceDurations & { slotType: SlotType };
 
-  const attendanceTableData = Object.entries(customerAttendance).map(
-    formatToTableData(customers)
-  );
-  attendanceTableData.sort(compareTableLines);
+type Attendance = NonNullable<LocalStore["firestore"]["data"]["attendance"]>;
+type Customers = NonNullable<LocalStore["firestore"]["data"]["customers"]>;
+type Slots = Record<string, SlotInterface>;
 
-  return attendanceTableData;
+export const processAttendances = (
+  attendance: Attendance,
+  slots: Slots,
+  customers: Customers,
+  month: string
+) => {
+  const final = wrapIter(Object.entries(attendance))
```

Comment:
Is it too much to turn this into an arrow function and remove 
```
{
  const final =
```
and remove the final `return final;}`?